tests/phpunit/test-class-wp-widget-media.php

```diff
@@ -0,0 +1,354 @@
+			'post_mime_type' => 'image/jpeg',
+		) );
+
+		$wp_widget_media = new ReflectionClass( 'WP_Widget_Media' );
+		$create_link_for = $wp_widget_media->getMethod( 'create_link_for' );
+		$create_link_for->setAccessible( true );
+
+		$result = $create_link_for->invokeArgs( $this->get_mocked_class_instance(), array(
+			get_post( $attachment_id ),
+		) );
+		$this->assertSame( '<a href="#"></a>',$result );
+
+		wp_update_post( array(
+			'ID' => $attachment_id,
+			'post_title' => 'Attachment Title',
+		) );
+
+		$result = $create_link_for->invokeArgs( $this->get_mocked_class_instance(), array(
+			get_post( $attachment_id ),
+			'file',
+		) );
+		$this->assertSame( '<a href="' . esc_url( wp_get_attachment_url( $attachment_id ) ) . '">Attachment Title</a>',$result );
+
+		$result = $create_link_for->invokeArgs( $this->get_mocked_class_instance(), array(
+			get_post( $attachment_id ),
+			'post',
+		) );
+		$this->assertSame( '<a href="http://example.org/?attachment_id=4">Attachment Title</a>',$result );
+	}
+
+	/**
+	 * Test widget method.
+	 *
+	 * @covers WP_Widget_Media::widget()
+	 * @covers WP_Widget_Media::render_media()
+	 */
+	function test_widget() {
+		$args     = array(
+			'before_title'  => '<h2>',
+			'after_title'   => "</h2>\n",
+			'before_widget' => '<section>',
+			'after_widget'  => "</section>\n",
+		);
+		$instance = array(
+			'title'         => 'Foo',
+			'url'           => '',
+			'attachment_id' => 0,
+		);
+
+		ob_start();
+		$widget = $this->get_mocked_class_instance();
+		$widget->expects( $this->atLeastOnce() )->method( 'render_media' )->with( $instance );
+		$widget->widget( $args, $instance );
+		$output = ob_get_clean();
+
+		$this->assertContains( '<h2>Foo</h2>', $output );
+		$this->assertContains( '<section>', $output );
+		$this->assertContains( '</section>', $output );
+
+		// No title.
+		ob_start();
+		$widget            = $this->get_mocked_class_instance();
+		$instance['title'] = '';
+		$widget->expects( $this->atLeastOnce() )->method( 'render_media' )->with( $instance );
+		$widget->widget( $args, array() );
+		$output = ob_get_clean();
+		$this->assertNotContains( '<h2>Foo</h2>', $output );
+	}
+
+	/**
+	 * Test form method.
+	 *
+	 * @covers WP_Widget_Media::form()
+	 */
+	function test_form() {
+		$widget = $this->get_mocked_class_instance();
+
+		ob_start();
+		$widget->form( array() );
+		$output = ob_get_clean();
+
+		$this->assertContains( 'name="widget-mocked[][attachment_id]"', $output );
+		$this->assertContains( 'name="widget-mocked[][title]"', $output );
+		$this->assertContains( 'name="widget-mocked[][url]"', $output );
+	}
+
+	/**
+	 * Test display_media_state method.
+	 *
+	 * @covers WP_Widget_Media::display_media_state()
+	 */
+	function test_display_media_state() {
+		$widget        = $this->get_mocked_class_instance();
+		$attachment_id = self::factory()->attachment->create_object( DIR_TESTDATA . '/images/canola.jpg', 0, array(
```

Comment:
Passing a string as the first argument is deprecated. The function signature for `create_object` is:

```
/**
 * Create an attachment fixture.
 *
 * @param array $args {
 *     Array of arguments. Accepts all arguments that can be passed to
 *     wp_insert_attachment(), in addition to the following:
 *     @type int    $post_parent ID of the post to which the attachment belongs.
 *     @type string $file        Path of the attached file.
 * }
 * @param int   $legacy_parent Deprecated.
 * @param array $legacy_args   Deprecated
 */
```